clients/v1/clients.proto

```diff
@@ -102,6 +118,54 @@ message Role {
   optional string resource_type = 4;
 }
 
+message GetUserMetadataRequest {
+  // The user's external ID.
+  string id = 1;
+  // Services is the list of service slugs to filter the metadata. "*" can be used
+  // to return metadata for all services.
+  //
+  // Requires the scope for the requested service(s); for example, '*' requires
+  // 'sams::user.metadata::read', and 'dotcom' requires 'sams::user.metadata.dotcom::read'.
+  repeated string services = 2;
+}
+
+// The metadata associated with a user for a specific service.
+message UserServiceMetadata {
+  // The user's external ID.
+  //
+  // Behaviour: IDENTIFIER
+  string user_id = 1;
+  // The service slug.
+  //
+  // Behaviour: IDENTIFIER
+  string service = 2;
+  // Metadata associated with this user for the specified service.
+  map<string, string> metadata = 3;
```

Comment:
In SAMS, the value type is not strictly `string`, I think it's fine to be stricter here if it's too much trouble to support `any`.

![CleanShot 2024-12-09 at 18 00 29@2x](https://github.com/user-attachments/assets/b9968380-2195-4cc6-acb5-d6b57ceabffb)

Reply:
ah, yes we can support a more arbitrary shape with https://github.com/protocolbuffers/protobuf/blob/main/src/google/protobuf/struct.proto, we've used this for telemetry